File: app/src/main/java/net/rmoreno/weatherapp/WeatherInteractor.java
package net.rmoreno.weatherapp;


import com.squareup.okhttp.Callback;
import com.squareup.okhttp.Request;
import com.squareup.okhttp.Response;

import net.rmoreno.weatherapp.presenters.Presenter;
import net.rmoreno.weatherapp.repositories.WeatherRepository;

import java.io.IOException;

public class WeatherInteractor {

    private WeatherRepository repo = new WeatherRepository();

    public WeatherInteractor(WeatherRepository repo) {
        this.repo = repo;
    }

    public void getWeatherData(double lat, double lng, final Presenter.WeatherCallback weatherCallback) {
        repo.getCurrentWeather(lat, lng, new Callback() {
            @Override
            public void onFailure(Request request, IOException e) {
                weatherCallback.onFailure(request, e);
            }

            @Override
            public void onResponse(Response response) throws IOException {
                weatherCallback.onWeatherRetrieved(response);
            }
        });
    }


}

File: app/src/main/java/net/rmoreno/weatherapp/WeatherInteractor.kt
package net.rmoreno.weatherapp

import android.location.Location
import com.google.android.gms.tasks.Task
import io.reactivex.Observable
import net.rmoreno.weatherapp.models.ForecastResponse
import net.rmoreno.weatherapp.repositories.WeatherRepository

class WeatherInteractor(var repo: WeatherRepository, var locationSensor: LocationSensor) {
    fun getWeatherData(lat: Double, lng: Double): Observable<ForecastResponse> {
        return repo.getForecast(lat, lng)
    }

    fun getCurrentLocation(): Task<Location> {
        return locationSensor.getLastLocation()
    }

    val sweaterWeather: Int
        get() = repo.sweaterTemp
}
